msupply-foundation/open-msupply
# 3503 mass reduce to zero

Fixes #3503

# 👩🏻‍💻 What does this PR do? 
 

From stocktake, adds ability to mass reduce to 0, and mass change location:

Reason input will only show if reasons are required:
![Screenshot 2024-04-11 at 5 25 44 PM](https://github.com/msupply-foundation/open-msupply/assets/55115239/c7e0f93d-1edf-4167-927b-98954911da20)

![Screenshot 2024-04-11 at 5 25 51 PM](https://github.com/msupply-foundation/open-msupply/assets/55115239/77514511-f5a5-48de-8e4f-7400a7ca2e40)


# 🧪 How has/should this change been tested? 


- Create a stocktake
- Add some lines
- Select some lines
- Select action "reduce lines to 0"
- [ ] Modal appears - if reasons configured, you have to enter a reason
- [ ] Ok button enables - click ok
- [ ] Selected placeholder lines (blue) are now included lines (black), the counted # packs is 0, and your reason should be applied to all of the selected lines
- Select some lines
- Select action "change location of lines"
- [ ] Modal appears - select a new location
- [ ] Click ok - all selected lines should now have their location updated



## 💌 Any notes for the reviewer?


See inline comments

## 📃 Documentation


- [ ] From stocktake - how to mass reduce lines
- [ ] From stocktake - how to change location
